exchange-contract/README.md

```diff
@@ -0,0 +1,79 @@
+# README
+
+This repository contains a prototype implementation of protocols that could be used for enabling Non Fungible Tokens (NFTs) for confidential assets. The protocols and software are for reference purposes only and not intended for production usage.
+
+## Problem Statement
```

Comment:
Structural: I would move this section and the next two sections into a doc/intro.md file for better legibility of the README, but link to that doc from here.

Reply:
Significant overhaul in documentation. Please re-review.